android/app/src/main/kotlin/com/jabook/app/jabook/audio/processors/LufsAnalysisWorker.kt

```
@@ -0,0 +1,332 @@
+        }
+
+        /**
+         * Performs distributed LUFS estimation on a single audio file.
+         *
+         * Reads [SAMPLE_WINDOWS] evenly-spaced chunks and integrates the results
+         * using energy-weighted averaging via [LufsRmsEstimator.integrateEstimates].
+         *
+         * Uses [MediaCodec] to decode compressed audio formats (MP3, AAC, etc.) into
+         * PCM before estimating LUFS, ensuring accurate loudness measurement.
+         */
+        internal fun estimateLufsForFile(audioFile: File): Double? {
```

detekt (1.23.8)

[warning] The function estimateLufsForFile appears to be too complex based on Cognitive Complexity (complexity: 43). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)